### PR TITLE
Change go.sh to generic version

### PR DESCRIPTION
**Pull request by xiulipan**
Read git branch info for version number. No need to update
go.sh for each release.

Signed-off-by: Pan Xiuli <xiuli.pan@linux.intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
Can you please look at the review comments in PR #10?

**Comment by xiulipan** [comment]
@marc-hb What comment you mean here? Do you mean the following?

> This should be two `for` loops:
> 
> ```shell
> for product in bdw byt cht; do
>     ln -s ${VERSION}/sof-${product}-${VERSION}.ri sof-${product}.ri
> done
> ```

And 

> Could you move all the code into a `main()` function? It makes it easier to source the script for interactive testing by commenting out just the final line, lets add functions in any order and moves everything at the same indentation level (when this script grows functions).
> 
> ```shell
> # stop on most errors
> set -e
> 
> main()
> {
>   # all the code 
> }
> main "$@"
> ```
> 
> Also please run `shellcheck` on the script, it will show a few quoting issues.

**Comment by plbossart** [comment]
the plan was to move to a Makefile and drop this file.

File: go.sh [hunk 1]
```diff
@@ -20,5 +20,36 @@
 #    files (e.g. signed public releases should match signed intel releases).
 #    "publish" rule. 
 
-echo "Nothing implemented"
+ROOT=
+INTEL_PATH=lib/firmware/intel
+VERSION=$(git symbolic-ref HEAD 2>/dev/null | cut -d"/" -f 3| cut -d"-" -f 2)
```
Comment (marc-hb) [inline go.sh:25]: git branch --show-current

**Comment by maaarghk** [comment]
aah, 

1) unimplemented file is still in 1.6 branch
2) I'm having a look at the arch package as I want to update to 1.6 and I note it downloads https://github.com/thesofproject/sof-bin/archive/stable-v$pkgver.tar.gz rather than doing a git checkout, therefore `git branch --show-current` method will not work easily. Maybe that is not a problem for you but just raising it :)

**Comment by lgirdwood** [comment]
@marc-hb are you able to look at removing go.sh whilst @xiulipan and the rest of PRC team are on vacation.

> 2\. I'm having a look at the arch package as I want to update to 1.6 and I note it downloads https://github.com/thesofproject/sof-bin/archive/stable-v$pkgver.tar.gz rather than doing a git checkout, therefore `git branch --show-current` method will not work easily. Maybe that is not a problem for you but just raising it :)

@maaarghk I'm not familiar with the arch package. Can you point me to it ? Comments are welcome as this is rc3 and we are trying to improve the deployment for publishers prior to v1.6 final.

**Comment by maaarghk** [comment]
Hi @lgirdwood - it is here:

https://github.com/archlinux/svntogit-packages/blob/packages/sof-firmware/trunk/PKGBUILD

prepare() runs and modifies go.sh with correct variables.  then `source`s are downloaded and verified. then package() runs go.sh with the appropriate root for the arch build system.

**Comment by lgirdwood** [comment]
> Hi @lgirdwood - it is here:
> 
> https://github.com/archlinux/svntogit-packages/blob/packages/sof-firmware/trunk/PKGBUILD
> 
> prepare() runs and modifies go.sh with correct variables. then `source`s are downloaded and verified. then package() runs go.sh with the appropriate root for the arch build system.
> 
> (sorry, I didn't know it was rc3, I never noticed any indication about 1.6 not being the final release).

ok, thanks - we are planning on replacing go.sh with a Makefile but lets keep it atm so folks dont have to manually copy stuff.  @marc-hb I've no idea if you will have time to remove go.sh and replace with a Makefile prior to v1.6 final in next two weeks ?

**Comment by marc-hb** [comment]
I'm not even at the stage where I understand everything that needs to be done here and I have some other relatively high priority stuff to do, so I don't think I can commit to a 2 weeks deadline, sorry.

**Comment by lgirdwood** [comment]
@marc-hb dont worry, your updates are needed for v1.7 so you have more than 2 weeks !

**Pull request closed without merging**